x/transferhook/types/codec.go

```diff
@@ -17,6 +22,18 @@ func RegisterInterfaces(registry cdctypes.InterfaceRegistry) {
 }
 
 var (
-	Amino     = codec.NewLegacyAmino()
-	ModuleCdc = codec.NewAminoCodec(Amino)
+	amino     = codec.NewLegacyAmino()
+	ModuleCdc = codec.NewAminoCodec(amino)
 )
+
+func init() {
+	RegisterCodec(amino)
+	cryptocodec.RegisterCrypto(amino)
+	sdk.RegisterLegacyAminoCodec(amino)
+
+	// Register all Amino interfaces and concrete types on the authz  and gov Amino codec so that this can later be
+	// used to properly serialize MsgGrant, MsgExec and MsgSubmitProposal instances
+	RegisterCodec(authzcodec.Amino)
+	RegisterCodec(govcodec.Amino)
+	RegisterCodec(groupcodec.Amino)
```

Comment:
why do we require these 3 in every module ?

Reply:
only 1 should be enough, I think ?

Reply:
These three lines register each module's msgs in the amino codec of authz, gov, and group modules. These 3 modules require messages from all the modules in the chain to marshall and unmarshal the messages when received at their end. Authz to grant permissions and execute msgs, gov to add proposals (from v0.47.x) any tx can be executed using gov proposal and group module I am not aware of but pretty some similar reason